lib/new_relic/environment_report.rb

```diff
@@ -71,7 +71,13 @@ def self.registered_reporters=(logic)
     report_on('OS'                ) { ::NewRelic::Agent::SystemInfo.ruby_os_identifier     }
     report_on('Database adapter'  ) do
       begin
-        ActiveRecord::Base.configurations.configs_for(env_name: NewRelic::Control.instance.env, spec_name: "primary").config['adapter']
+        if ::ActiveRecord::Base.respond_to?(:connection_db_config)
+          ActiveRecord::Base.configurations.configs_for(env_name: NewRelic::Control.instance.env, name: "primary")
+            .connection_db_config.configuration_hash['adapter']
+        else
+          ActiveRecord::Base.configurations.configs_for(env_name: NewRelic::Control.instance.env, spec_name: "primary").config['adapter']
+        end
+
       rescue NoMethodError
```

Comment:
Incidentally, your PR brought to light another potential improvement: the `NoMethodError` that surfaced in my testing was caught here and raised another deprecation warning:

```
DEPRECATION WARNING: [] is deprecated and will be removed from Rails 6.2 (Use configs_for) (called from rescue in block in <class:EnvironmentReport> at /Users/rklein/newrelic/ruby_agent/lib/new_relic/environment_report.rb:82)
```

If you're interested in addressing this as well, you'd be most welcome. Otherwise, I can add it to our backlog for Rails 6.1 support.

Reply:
I will address this also.